csvapi/parser.py

```diff
@@ -28,20 +28,21 @@ def from_csv(filepath, encoding='utf-8', sniff_limit=SNIFF_LIMIT):
         return agate.Table.from_csv(filepath, encoding=encoding, column_types=agate_tester())
 
 
-def from_excel(filepath):
-    import agateexcel  # noqa
-    return agate.Table.from_xls(filepath, column_types=agate_tester())
-
-
 def to_sql(table, urlhash, storage):
     db_info = get_db_info(urlhash, storage=storage)
     table.to_sql(db_info['dsn'], db_info['db_name'], overwrite=True)
 
 
 def parse(filepath, urlhash, storage, encoding=None, sniff_limit=SNIFF_LIMIT):
-    if is_binary(filepath):
-        table = from_excel(filepath)
-    else:
+    import agateexcel  # noqa
```

Comment:
I don't remember why I did not do a top import. Can you test with a top import? If it causes problems, this might be why we had a `from_excel` fn :-)